forge-gui-mobile/src/forge/adventure/data/AdventureQuestData.java

```diff
@@ -437,8 +520,49 @@ public void updateEventComplete(AdventureEventData completedEvent) {
         for (AdventureQuestStage stage: stages) {
             if(failed)
                 break;
+            if (stage.getStatus() != Active)
+                continue;
             stage.updateEventComplete(completedEvent);
-            failed = failed || stage.getStatus() == AdventureQuestController.QuestStatus.Failed;
+            failed = failed || stage.getStatus() == Failed;
+        }
+        if (!failed)
+            updateStatus();
+    }
+
+    public void updateQuestComplete(AdventureQuestData completedQuest) {
+        for (AdventureQuestStage stage: stages) {
+            if(failed)
+                break;
+            if (stage.getStatus() != Active)
+                continue;
+            stage.updateQuestComplete(completedQuest);
+            failed = failed || stage.getStatus() == Failed;
```

Comment:
As of right now, no. But there's room for it in the future.

Quest 1: Ally with X world entity instead of Y
Quest 2: Ally with Y world entity instead of X
If quest 1 is completed, quest 2 fails or vice versa.

This will be a bit less overtly awkward when all the update methods are cleaned up a bit as per your next suggestion.